src/views/TodoList/TodoList.js

```diff
@@ -0,0 +1,74 @@
+import { useState, useEffect } from 'react';
+import { fetchTodos, createTodo, updateTodo } from '../../services/todos';
+import './TodoList.css';
+
+export default function TodoList() {
+  const [todos, setTodos] = useState([]);
+  const [description, setDescription] = useState('');
+  const [error, setError] = useState('');
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const todosData = await fetchTodos();
+        setTodos(todosData);
+        setLoading(false);
+      } catch (e) {
+        setError('An error has occurred, please try again.');
+      }
+    };
+    fetchData();
+  }, []);
+
+  const handleSubmit = async () => {
+    try {
+      const newTodo = await createTodo({ description, complete: false });
+      setTodos((prevState) => [...prevState, newTodo]);
+    } catch (e) {
+      setError('An error has occurred, please try again');
+    }
+  };
+
+  const handleCheckbox = async (todo) => {
+    try {
+      if (todo.complete === false) {
+        await updateTodo({ id: todo.id, complete: true });
+      } else {
+        await updateTodo({ id: todo.id, complete: false });
+      }
```

Comment:
Here we can also call await updateTodo({ id: todo.id, complete: !todo.complete }) to render the opposite of the truthy or falsy value that is currently set.